chemprop/v2/cli/train.py

```diff
@@ -0,0 +1,414 @@
+    parser.add_argument(
+        "-c",
+        "--n-cpu",
+        "--num-workers",
+        type=int,
+        default=0,
+        help="the number of CPUs over which to parallelize data loading",
+    )
+
+    return parser
+
+
+def process_args(args: Namespace):
+    args.input = Path(args.input)
+    args.output_dir = Path(args.output_dir or Path.cwd() / args.input.stem)
+    args.logdir = Path(args.logdir or args.output_dir / "logs")
+
+    args.output_dir.mkdir(exist_ok=True, parents=True)
+    args.logdir.mkdir(exist_ok=True, parents=True)
+
+
+def validate_args(args):
+    pass
+
+
+def main(args):
+    bond_messages = not args.atom_messages
+    n_components = len(args.smiles_columns)
+    n_tasks = len(args.target_columns)
+    bounded = args.loss_function is not None and "bounded" in args.loss_function
+
+    if n_components > 1:
+        warnings.warn(
+            "Multicomponent input is not supported at this time! Using only the 1st input..."
+        )
+
+    format_kwargs = dict(
+        no_header_row=args.no_header_row,
+        smiles_columns=args.smiles_columns,
+        target_columns=args.target_columns,
+        bounded=bounded,
+    )
+    featurization_kwargs = dict(
+        features_generators=args.features_generators,
+        explicit_h=args.explicit_h,
+        add_h=args.add_h,
+        reaction=0 in args.rxn_idxs,
+    )
+
+    all_data = build_data_from_files(
+        args.input,
+        **format_kwargs,
+        p_features=args.features_path,
+        p_atom_feats=args.atom_features_path,
+        p_bond_feats=args.bond_features_path,
+        p_atom_descs=args.atom_descriptors_path,
+        **featurization_kwargs,
+    )
+
+    if args.val_path is None and args.test_path is None:
+        train_data, val_data, test_data = split_data(all_data, args.split, args.split_sizes)
+    elif args.test_path is not None:
+        test_data = build_data_from_files(
+            args.test_path,
+            p_features=args.test_features_path,
+            p_atom_feats=args.test_atom_features_path,
+            p_bond_feats=args.test_bond_features_path,
+            p_atom_descs=args.test_atom_descriptors_path,
+            **format_kwargs,
+            **featurization_kwargs,
+        )
+        if args.val_path is not None:
+            val_data = build_data_from_files(
+                args.val_path,
+                p_features=args.val_features_path,
+                p_atom_feats=args.val_atom_features_path,
+                p_bond_feats=args.val_bond_features_path,
+                p_atom_descs=args.val_atom_descriptors_path,
+                **format_kwargs,
+                **featurization_kwargs,
+            )
+            train_data = all_data
+        else:
+            train_data, val_data, _ = split_data(all_data, args.split, args.split_sizes)
+    else:
+        raise ArgumentError("'val_path' must be specified is 'test_path' is provided!")
```

Comment:
I think this should be:  ... must be specified if ...
Making a comment so that me or someone else will remember to fix that later.